cinema/tests/test_movie_api.py

```diff
@@ -157,3 +158,147 @@ def test_image_url_is_shown_on_movie_session_detail(self):
+
+
+class UnauthenticatedMovieApiTest(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_auth_required(self):
+        res = self.client.get(MOVIE_URL)
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class AuthenticatedMovieApiTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            "test@test.com",
+            "testpass"
+        )
+        self.client.force_authenticate(self.user)
+
+    def test_list_movies(self):
+        sample_movie()
+        movie_with_genre = sample_movie()
+
+        genre1 = sample_genre(name="comedy")
+        genre2 = sample_genre(name="romance")
+
+        actor1 = sample_actor(first_name="test1")
+        actor2 = sample_actor(first_name="test2")
+
+        movie_with_genre.genres.add(genre1, genre2)
+        movie_with_genre.actors.add(actor1, actor2)
+
+        res = self.client.get(MOVIE_URL)
+
+        movies = Movie.objects.all()
+        serializer = MovieListSerializer(movies, many=True)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_filter_movies_by_title(self):
+        movie1 = sample_movie(title="A")
+        movie2 = sample_movie(title="B")
+
+        genre1 = sample_genre(name="comedy")
+        genre2 = sample_genre(name="romance")
+
+        actor1 = sample_actor(first_name="test1")
+        actor2 = sample_actor(first_name="test2")
+
+        movie1.genres.add(genre1)
+        movie2.genres.add(genre2)
+        movie1.actors.add(actor1)
+        movie2.actors.add(actor2)
+
+        movie3 = sample_movie(title="without genres adn actors")
+
+        res = self.client.get(MOVIE_URL, {
+            "genres": f"{genre1.id}, {genre2.id}",
+            "actors": f"{actor1.id}, {actor2.id}"
+        })
+
+        serializer1 = MovieListSerializer(movie1)
+        serializer2 = MovieListSerializer(movie2)
+        serializer3 = MovieListSerializer(movie3)
+
+        self.assertIn(serializer1.data, res.data)
+        self.assertIn(serializer2.data, res.data)
+        self.assertNotIn(serializer3.data, res.data)
+
+    def test_retrieve_movie_detail(self):
+        movie = sample_movie()
+        movie.genres.add(sample_genre())
+        movie.actors.add(sample_actor())
+
+        url = detail_url(movie.id)
+        res = self.client.get(url)
+
+        serializer = MovieDetailSerializer(movie)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_creat_movie_forbidden(self):
+        payload = {
+            "title": "Sample movie",
+            "description": "Sample description",
+            "duration": 90,
+        }
+        res = self.client.post(MOVIE_URL, payload)
+        self.assertEqual(res.status_code, status.HTTP_403_FORBIDDEN)
+
+
+class AdminApiTest(TestCase):
```

Comment:
would be nice to test that deletion of the movie is not allowed even for admin:)